web/modules/custom/joinup_invite/joinup_invite.module

```diff
@@ -58,3 +63,70 @@ function joinup_invite_block_view_alter(array &$build, BlockPluginInterface $blo
     $build['#cache']['contexts'] = Cache::mergeContexts($cache_contexts, ['og_role']);
   }
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_delete().
+ *
+ * If a membership is deleted and an invitation exists for the same group and
+ * user, delete the invitation as well.
+ */
+function joinup_invite_og_membership_delete(OgMembershipInterface $entity) {
+  $invitations = \Drupal::entityTypeManager()->getStorage('invitation')->loadByProperties([
+    'entity_type' => $entity->getGroupEntityType(),
+    'entity_id' => $entity->getGroupId(),
+    'recipient_id' => $entity->getOwnerId(),
+    'bundle' => 'group',
+  ]);
+
+  // Normally, only one invitation should exist per entity and user, but iterate
```

Comment:
This no more true Now we can have a lot of rejected invitations for the same user and entity.